src/routes/versions.rs

```diff
@@ -63,17 +66,47 @@ pub async fn versions_get(
         .await
         .map_err(|e| ApiError::DatabaseError(e.into()))?;
 
-    let versions: Vec<models::mods::Version> = versions_data
-        .into_iter()
-        .filter_map(|v| v)
-        .map(convert_version)
-        .collect();
+    let mut versions = Vec::new();
+
+    for version_data in versions_data {
+        if let Some(version) = version_data {
+            let mut authorized = version.accepted;
+
+            if !authorized {
+                let user = get_user_from_headers(req.headers(), &**pool).await?;
+                let mod_item = database::models::Mod::get(version.mod_id, &**pool)
+                    .await
+                    .map_err(|e| ApiError::DatabaseError(e.into()))?
+                    .ok_or_else(|| {
+                        ApiError::InvalidInputError(
+                            "Attempted to edit version not attached to mod. How did this happen?"
+                                .to_string(),
+                        )
+                    })?;
+                let team_member = database::models::TeamMember::get_from_user_id(
+                    mod_item.team_id,
+                    user.id.into(),
+                    &**pool,
+                )
+                .await?;
+
+                authorized = team_member.is_some()
+                    || user.role == Role::Moderator
+                    || user.role == Role::Admin
```

Comment:
This should be moved up to the definition of `authorized` so that it doesn't have to query the database for every unapproved mod for moderators / admins.